src/v/net/tests/conn_quota_test.cc

```diff
@@ -253,6 +253,7 @@ void conn_quota_fixture::test_borrows(
     // Now that all units are released, we should find that reclaim
     // flag is switched off everywhere.
     vlog(logger.debug, "Checking reclaim status");
+    ss::thread::yield(); // give the backgrounded part of reclaim_to a chance
```

Comment:
motivation here makes sense, though it's possible that it needs to be in a loop? my reasoning is that on a release build with fifo scheduling it might be deterministic, but on debug builds seastar will randomize the schedule.

Reply:
I looked again at what happens when we yield, and in this instance because we're waiting for things to execute on other cores, no amount of explicit yielding on one core is sufficient -- changed this to a cooperative_spin_wait